include/motor-control/core/stepper_motor/motor_interrupt_handler.hpp

```diff
@@ -283,6 +283,10 @@ class MotorInterruptHandler {
         if (limit_switch_triggered()) {
             position_tracker = 0;
             hardware.reset_step_tracker();
+            if (stall_checker.has_encoder()) {
```

Comment:
I don't think we should be setting the encoder flag here, and we should also probably get rid of the `hardware.reset_step_tracker()` line above it. Since we define the end of our homing sequence to be when the motors are backed off of the switches, we should only reset all of the trackers after backing off of the switch.

(If this is here because the hardware-controller expects the encoder flag to be asserted then we should change in that in Python imo)